### PR TITLE
emacs.pkgs.seq: stop shadowing it

### LINKED ISSUE #272019: Emacs Magit doesn’t work on 23.11
### Describe the bug
When on tries to open magit on 23.11 you get the `^L` in the output and the commands aren’t working as expected.

### Steps To Reproduce
1. Update to 23.11
2. Start Emacs and Magit in a Git repo of your choice.
3. Try to select a commit for example.

### Expected behavior
The commit diff should be shown.

### Screenshots
image

### Additional context
The issue is already fixed upstream but requires some changes in Nixpkgs:
- https://github.com/magit/magit/issues/5011
- https://github.com/magit/magit/issues/5011#issuecomment-1836861539

### Notify maintainers

@AndersonTorres
@adisbladis
@Atemu 
@jwiegley
@lovek323
@matthewbauer

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
- system: `"x86_64-linux"`
 - host os: `Linux 6.1.64, NixOS, 23.11 (Tapir), 23.11.20231202.933d7dc`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `""`
 - nixpkgs: `/nix/store/chi01x4fh8hx7wwh0rqd7m1c8583iyhv-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by jian-lin**
## Description of changes

magit requires[1] seq 2.24.  seq from GNU Elpa satisfies that.
However, it is shadowed by the Emacs builtin one to workaround an old
bug[2] and the version of the builtin seq in Emacs 28 is only 2.23.
So magit is broken for Emacs 28 which is the default one in NixOS
23.11 and available for the unstable branch.
    
This patch fixes magit by stopping shadowing seq from GNU Elpa since
that old bug[2] is not relevant now.
    
Fixes https://github.com/NixOS/nixpkgs/issues/272019.
    
[1]: https://github.com/magit/magit/blob/f4ff817cb2a48f0f7887050c3be469c03a059567/lisp/magit.el#L27
[2]: https://github.com/NixOS/nixpkgs/pull/74936

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jian-lin** [comment]
@ofborg eval

**Comment by jian-lin** [comment]
@ofborg eval

File: pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix [hunk 1]
```diff
@@ -60,12 +60,6 @@ self: let
         else super.org-transclusion;
       rcirc-menu = markBroken super.rcirc-menu; # Missing file header
       cl-lib = null; # builtin
-      cl-print = null; # builtin
```
Comment (adisbladis) [inline pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix:63]: Don't "clean up" old declarations. These are still useful for tools like https://github.com/nix-community/emacs-overlay?tab=readme-ov-file#extra-library-functionality and nix-doom-emacs.

**Review by adisbladis — DISMISSED** [review]

File: pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix (hunk 1 above)
Comment (jian-lin) [reply pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix:63]: > These are still useful for tools like https://github.com/nix-community/emacs-overlay?tab=readme-ov-file#extra-library-functionality and nix-doom-emacs.

Could you elaborate why they are useful?

`rg` finds nothing for `tle` and `advice` in Nixpkgs, emacs-overlay and nix-doom-emacs.

While there are references for `cl-print`, they are all [guarded][1] by `? null`.  I am not sure how that `? null` is generated, so deleting `cl-print` may cause issues.

To sum up, I think it is safe to delete `tle` and `advice`.  I am also fine with keeping all three of them.  WDYT?

[1]: https://github.com/NixOS/nixpkgs/blob/ba195b10cc77767b2b2ac6e01fb7045f8d29f137/pkgs/applications/editors/emacs/elisp-packages/elpa-generated.nix#L1696

Comment (jian-lin) [reply pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix:63]: That commit is removed and now all three of them remain unchanged.

**Comment by jian-lin** [comment]
I do not find any regression with this patch applied to my system.

CI passed before.  I just fixed a typo in the commit message and force-pushed them.  So I will not wait for CI again.
